src/index.ts

```diff
@@ -0,0 +1,275 @@
+	}
+}
+
+function semverCompare(version1: string, version2 = "3.4.0") {
+	const version1Parts = version1.split(".");
+	const version2Parts = version2.split(".");
+
+	for (const version1Part of version1Parts) {
+		const version2Part = version2Parts.shift();
+
+		if (version1Part !== version2Part && version2Part) {
+			return version1Part < version2Part ? true : false;
+		}
+	}
+
+	return false;
+}
+
+async function uploadSecrets() {
+	startGroup("🔑 Uploading Secrets");
+	const secrets: string[] = config["secrets"];
+	const environment = config["ENVIRONMENT"];
+	const workingDirectory = config["workingDirectory"];
+
+	if (!secrets.length) {
+		return;
+	}
+	if (semverCompare(config["WRANGLER_VERSION"]))
+		return legacyUploadSecrets(secrets, environment, workingDirectory);
+
+	const secretObj = secrets.reduce((acc: any, secret: string) => {
+		acc[secret] = getSecret(secret);
+		return acc;
+	}, {});
+
+	const environmentSuffix = !environment.length ? "" : ` --env ${environment}`;
+
+	const secretCmd = `echo ${JSON.stringify(
+		secretObj,
+	)} | ${getNpxCmd()} wrangler secret:bulk ${environmentSuffix}`;
+
+	try {
+		execSync(secretCmd, {
+			cwd: workingDirectory,
+			env: process.env,
+			stdio: "ignore",
+		});
+		info(`✅ Uploaded secrets`);
+	} catch {
+		setFailed(`🚨 Error uploading secrets`);
+	}
+
+	endGroup();
+}
+
+function defaultCommandBehavior() {
+	const environment = config["ENVIRONMENT"];
+	const wranglerVersion = config["WRANGLER_VERSION"];
+	const vars = config["VARS"];
+	const workingDirectory = config["workingDirectory"];
+
+	const deployCommand =
+		wranglerVersion === "latest" || wranglerVersion.startsWith("3")
+			? "deploy"
+			: "publish";
+
+	info(
+		`📌 No Wrangler commands were provided, executing default deployment. '${deployCommand}'`,
+	);
+
+	const envVarArray = vars.map((envVar: string) => {
+		if (process.env[envVar] && process.env[envVar]?.length !== 0) {
+			return `${envVar}:${process.env[envVar]!}`;
+		} else {
+			setFailed(`🚨 ${envVar} not found in variables.`);
+		}
+	});
+
+	const envVarArg: string =
+		envVarArray.length > 0 ? `--var ${envVarArray.join(" ").trim()}` : "";
+
+	if (environment.length === 0) {
+		execSync(`${getNpxCmd()} wrangler ${deployCommand} ${envVarArg}`.trim(), {
+			cwd: workingDirectory,
+			env: process.env,
+		});
+	} else {
+		execSync(
+			`${getNpxCmd()} wrangler ${deployCommand} --env ${environment} ${envVarArg}`.trim(),
+			{ cwd: workingDirectory, env: process.env },
+		);
+	}
+}
+
+async function wranglerCommands() {
+	startGroup("🚀 Executing Wrangler Commands");
+	const commands = config["COMMANDS"];
+	if (!commands.length) {
+		defaultCommandBehavior();
+		return;
+	}
+	const environment = config["ENVIRONMENT"];
+
+	if (
+		environment.length > 0 &&
+		!commands.every((command) => command.includes(`--env ${environment}`))
+	) {
+		setFailed(
+			`🚨 Since an environment has been specified, each command must include '--env ${environment}'.`,
+		);
+	}
```

Comment:
If they specified an environment, then each command passed in needs to include an `--env` flag, thats not only how it worked previously but prevents the user from inadvertently passing a command thinking its doing to "dev" environment because they  put that in the Actions `environment: dev` and end up overwriting their production environment.

Reply:
why provide the `environment` input to the action at all then? what's stopping us from including `--env ${environment}` on each command we execute?

Reply:
Oh I see your point, we can just do it for them, yeah I can do that it will be a change in behavior but its a major version bump I am cool with it.